### ISSUE TITLE
TradeRules are not being applied

### ISSUE THREAD

**Issue by Hulkstance** [JKorf/Binance.Net#1268]
I've upgraded to v9.0.1 and looks like `socketClient.SpotOptions` doesn't actually set anything. I've put a breakpoint on `BinanceSocketClientSpotApi.CheckTradeRules` and ApiOptions remains the same (with default values).

> [2023-07-03 04:08:41.738] [INF] [7] [] [Proto.OneForOneStrategy] - "Restarting" "nonhost/$1" because of "System.InvalidOperationException: Failed to place a market order: Illegal characters found in parameter 'quantity'; legal range
 is '^([0-9]{1,20})(\\.[0-9]{1,20})?$'.\r\n   at

![image](https://github.com/JKorf/Binance.Net/assets/33224919/9e07b8de-39a4-4314-ac90-f5079e6ef077)

Here is my configuration:

```cs
services.AddBinance(restClient =>
{
    restClient.Environment = BinanceEnvironment.Live;

    restClient.ApiCredentials = new ApiCredentials(ApiKey, SecretKey);
    restClient.AutoTimestamp = true;
    restClient.TimestampRecalculationInterval = TimeSpan.FromMinutes(30);

    restClient.SpotOptions.RateLimitingBehaviour = RateLimitingBehaviour.Wait;
    restClient.SpotOptions.TradeRulesBehaviour = TradeRulesBehaviour.AutoComply;
    restClient.SpotOptions.TradeRulesUpdateInterval = TimeSpan.FromSeconds(30);
}, socketClient =>
{
    socketClient.Environment = BinanceEnvironment.Live;

    socketClient.ApiCredentials = new ApiCredentials(ApiKey, SecretKey);
    socketClient.AutoReconnect = true;
    socketClient.ReconnectInterval = TimeSpan.FromSeconds(15);

    socketClient.SpotOptions.TradeRulesBehaviour = TradeRulesBehaviour.AutoComply;
    socketClient.SpotOptions.TradeRulesUpdateInterval = TimeSpan.FromMinutes(30);
}, ServiceLifetime.Singleton);
```

## Edit:

The following works, so it's pretty much the extension methods.

```cs
public static IServiceCollection AddBinance(this IServiceCollection services)
{
    ArgumentNullException.ThrowIfNull(services);

    services.AddTransient<IBinanceRestClient, BinanceRestClient>(_ => new BinanceRestClient(options =>
    {
        options.Environment = BinanceEnvironment.Live;

        options.ApiCredentials = new ApiCredentials(ApiKey, SecretKey);
        options.AutoTimestamp = true;
        options.TimestampRecalculationInterval = TimeSpan.FromMinutes(30);

        options.SpotOptions.RateLimitingBehaviour = RateLimitingBehaviour.Wait;
        options.SpotOptions.TradeRulesBehaviour = TradeRulesBehaviour.AutoComply;
        options.SpotOptions.TradeRulesUpdateInterval = TimeSpan.FromSeconds(30);
    }));

    services.AddSingleton<IBinanceSocketClient, BinanceSocketClient>(_ => new BinanceSocketClient(options =>
    {
        options.Environment = BinanceEnvironment.Live;

        options.ApiCredentials = new ApiCredentials(ApiKey, SecretKey);
        options.AutoReconnect = true;
        options.ReconnectInterval = TimeSpan.FromSeconds(15);

        options.SpotOptions.TradeRulesBehaviour = TradeRulesBehaviour.AutoComply;
        options.SpotOptions.TradeRulesUpdateInterval = TimeSpan.FromMinutes(30);
    }));
```

**Comment by JKorf** [comment]
Thanks for the update, that made looking for the fix easier. I found the bug and fixed it in 9.0.2. Let me know if you have any more issues

**Comment by Hulkstance** [comment]
Thank you too! :) If I find something else, I would open a new ticket
